### PR TITLE
tree-sitter: update to 0.24.3, revbump of neovim, emacs, rizin.

### PR DESCRIPTION
**Pull request by dataCobra**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64-musl
  - i686

### CODE REVIEW

**Comment by tranzystorekk** [comment]
Why are the revbumps necessary? Judging by the shlib, the changes shouldn't be breaking.

File: common/shlibs [hunk 1]
```diff
@@ -4196,7 +4196,8 @@ libsword-1.8.1.so libsword-1.8.1_6
 libgivaro.so.9 givaro-4.1.1_1
 liblinbox.so.0 linbox-1.6.3_1
 libpari-gmp-tls.so.8 pari-2.15.0_1
-libtree-sitter.so.0 tree-sitter-0.19.0_1
+libtree-sitter.so.0 tree-sitter-0.24.3_1
+libtree-sitter.so.0.24 tree-sitter-0.24.3_1
```
Comment (tranzystorekk) [inline common/shlibs:4200]: the full soname shouldn't be necessary

Comment (dataCobra) [reply common/shlibs:4200]: While building `xbps-src` complained that the *.24 SONAME should also be part of `common/shlib`.

But I can check again.

Comment (dataCobra) [reply common/shlibs:4200]: I've tested again and building neovim fails with the following shlibs error:

```
=> neovim-0.10.2_2: running pre-pkg hook: 04-generate-runtime-deps ...
   SONAME: libluv.so.1 <-> libluv>=1.30.1.0_1
   SONAME: libvterm.so.0 <-> libvterm>=0.3_1
   SONAME: /usr/lib/lua/5.1/lpeg.so <-> lua51-lpeg>=1.1.0_2
   SONAME: libmsgpack-c.so.2 <-> msgpack>=1.1.0_1
   SONAME: libtree-sitter.so.0.24 <-> UNKNOWN PKG PLEASE FIX!
   SONAME: libunibilium.so.4 <-> unibilium>=2.0.0_1
   SONAME: libluajit-5.1.so.2 <-> LuaJIT>=2.0.0_1
   SONAME: libm.so.6 <-> glibc>=2.39_1
   SONAME: libuv.so.1 <-> libuv>=1.0.0_1
   SONAME: libc.so.6 <-> glibc>=2.39_1
=> ERROR: neovim-0.10.2_2: cannot guess required shlibs, aborting!
```

Comment (dataCobra) [reply common/shlibs:4199]: With amazing help from the IRC I've changed the common/shlib to use the correct value.

**Pull request merged**